test/integration/change-streams/change_stream.test.ts

```diff
@@ -1563,3 +1493,584 @@ describe('Change Streams', function () {
+              failCommands: ['getMore'],
+              errorCode: code
+            }
+          });
+
+          const changes = once(changeStream, 'change');
+          await once(changeStream.cursor, 'init');
+
+          await collection.insertOne({ name: 'bailey' });
+
+          const [change] = await changes;
+          expect(change).to.have.property('operationType', 'insert');
+
+          expect(aggregateEvents).to.have.lengthOf(2);
+        }
+      );
+    }
+
+    it(
+      'maintains the change stream options on resume',
+      { requires: { topology: '!single', mongodb: '>=4.2' } },
+      async function () {
+        changeStream = collection.watch([], changeStreamResumeOptions);
+
+        await client.db('admin').command(<FailPoint>{
+          configureFailPoint: is4_2Server(this.configuration.version)
+            ? 'failCommand'
+            : 'failGetMoreAfterCursorCheckout',
+          mode: { times: 1 },
+          data: {
+            failCommands: ['getMore'],
+            errorCode: resumableErrorCodes[0].code
+          }
+        });
+
+        expect(changeStream.cursor)
+          .to.have.property('options')
+          .that.containSubset(changeStreamResumeOptions);
+
+        const changes = once(changeStream, 'change');
+        await once(changeStream.cursor, 'init');
+
+        await collection.insertOne({ name: 'bailey' });
+
+        await changes;
+
+        expect(changeStream.cursor)
+          .to.have.property('options')
+          .that.containSubset(changeStreamResumeOptions);
+      }
+    );
+
+    context('when the error is not a resumable error', function () {
+      it(
+        'does not resume',
+        { requires: { topology: '!single', mongodb: '>=4.2' } },
+        async function () {
+          changeStream = collection.watch([]);
+
+          const unresumableErrorCode = 1000;
+          await client.db('admin').command(<FailPoint>{
+            configureFailPoint: is4_2Server(this.configuration.version)
+              ? 'failCommand'
+              : 'failGetMoreAfterCursorCheckout',
+            mode: { times: 1 },
+            data: {
+              failCommands: ['getMore'],
+              errorCode: unresumableErrorCode
+            }
+          });
+
+          const willBeError = once(changeStream, 'change').catch(error => error);
+          await once(changeStream.cursor, 'init');
+          await collection.insertOne({ name: 'bailey' });
+
+          const error = await willBeError;
+
+          expect(error).to.be.instanceOf(MongoServerError);
+          expect(aggregateEvents).to.have.lengthOf(1);
+        }
+      );
+    });
+  });
+
+  it(
+    'caches the server version after the initial aggregate call',
+    { requires: { topology: '!single' } },
+    async function () {
+      changeStream = collection.watch([], changeStreamResumeOptions);
+      await initIteratorMode(changeStream);
+
+      expect(changeStream.cursor.maxWireVersion).not.to.be.undefined;
```

Comment:
should this first check to make sure that the maxWireVersion is initially undefined? and that the subsequent version is a number?